tools/custom/raw-bytecode-analyser/README.md

```diff
@@ -0,0 +1,90 @@
+# Hedera Smart Contract Bytecode Analyser
+
+This Python script analyses the bytecode of Smart Contracts on the Hedera network to detect specific operand usages.
+It can help determine the likelihood that a Smart Contract creates a token using an SECP256k1 key.
+
+## Requirements
+
+- Python 3.6 or higher
+- `requests` library
+- `requests-cache` library: A transparent, persistent cache for the `requests` library to improve performance by caching HTTP responses.
+- `evmdasm` library: A tool for disassembling EVM bytecode to human-readable assembly instructions.
+
+To install the required Python libraries, run:
+
+```bash
+pip install requests requests-cache evmdasm
+```
+
+## Usage
+
+The script can be run from the command line with the following arguments:
+
+- `contract_id`: The ID of the smart contract whose bytecode you want to analyze.
+- `--previewnet`: Optional flag to use the Previewnet Mirror Node URL.
+- `--testnet`: Optional flag to use the Testnet Mirror Node URL.
+
+If no network flag is specified, the script defaults to using the Mainnet Mirror Node URL.
+
+### Basic Command
+
+```bash
+python detect_token_creation_with_secp_key.py <contract_id>
+```
+
+### Examples
+
+1. **Analyze a Contract on Mainnet**
+
+    ```bash
+    python detect_token_creation_with_secp_key.py 0.0.123456
+    ```
+
+2. **Analyze a Contract on Testnet**
+
+    ```bash
+    python detect_token_creation_with_secp_key.py 0.0.123456 --testnet
+    ```
+
+3. **Analyze a Contract on Previewnet**
+
+    ```bash
+    python detect_token_creation_with_secp_key.py 0.0.123456 --previewnet
+    ```
+
+## Features
+
+The script performs the following checks on the smart contract bytecode:
+
+- **External Calls Detection**: Identifies if there are any calls to external addresses.
+- **Specific Address Usage**: Checks for the usage of Hedera addresses 0x167.
+- **Function Selector Usage**: Detects if any of the token creating functions selectors are used.
```

Comment:
I would also mention here that these detections are an over-approximation, the analysis only looks for `PUSHn` instructions which not necessarily refer to these usages.

Reply:
https://github.com/hashgraph/hedera-smart-contracts/pull/761/files/7a83edc39b876d1709123d80d3ab2c7289ff3111#diff-80452d27f149976aaca8e1f4d80ed3d7df78fffb4ea2294dce6a2309cfd76096R70